src/wallet/api/wallet_manager.cpp

```diff
@@ -443,6 +446,28 @@ std::string WalletManagerImpl::resolveOpenAlias(const std::string &address, bool
     return addresses.front();
 }
 
+std::string WalletManager::checkUpdates(const std::string &software, const std::string &subdir, bool testnet)
+{
+#ifdef BUILD_TAG
+    static const char buildtag[] = BOOST_PP_STRINGIZE(BUILD_TAG);
+#else
+    static const char buildtag[] = "source";
+#endif
+
+    std::string version, hash;
+    MDEBUG("Checking for a new " << software << " version for " << buildtag);
+    if (!tools::check_updates(software, buildtag, testnet, version, hash))
+      return "";
+
+    if (tools::vercmp(version.c_str(), MONERO_VERSION) > 0)
+    {
+      std::string url = tools::get_update_url(software, subdir, buildtag, version);
+      MGINFO("Version " << version << " of " << software << " for " << buildtag << " is available: " << url << ", SHA256 hash " << hash);
+      return version + "|" + hash + "|" + url;
```

Comment:
It would, but it gets to Qt, and returning compound elements in Qt seems to be a massive PITA when I looked. But yes, it'd be better ideally.